kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/SchemaUtilsSuite.scala

```diff
@@ -33,6 +36,187 @@ class SchemaUtilsSuite extends AnyFunSuite {
       s"Error message '$msg' didn't contain: $shouldContain")
   }
 
+  //////////////////////
+  // isSuperset tests //
+  //////////////////////
+
+  private val field_a = new StructField("a", INTEGER, true)
+  private val field_b_1 = new StructField("1", INTEGER, true)
+  private val field_b_2 = new StructField("2", STRING, true)
+  private val field_b =
+    new StructField("b", new StructType(Arrays.asList(field_b_1, field_b_2)), true)
+  private val field_c_foobar_3 = new StructField("3", INTEGER, true)
+  private val field_c_foobar =
+    new StructField("foo.bar", new StructType(Arrays.asList(field_c_foobar_3)), true)
+  private val field_c = new StructField("c", new StructType(Arrays.asList(field_c_foobar)), true)
+  private val field_d = new StructField("d", new ArrayType(STRING, true), true)
+  private val field_e = new StructField("e", new MapType(INTEGER, STRING, true), true)
+  private val field_f_element_4 = new StructField("4", INTEGER, true)
+  private val field_f =
+    new StructField("f",
+      new ArrayType(new StructType(Arrays.asList(field_f_element_4)), true), true)
+  private val field_g_key_5 = new StructField("5", INTEGER, true)
+  private val field_g_value_zipzap_6 = new StructField("6", INTEGER, true)
+  private val field_g_value_zipzap_7 = new StructField("7", INTEGER, true)
+  private val field_g_value_zipzap =
+    new StructField("zip.zap",
+      new StructType(Arrays.asList(field_g_value_zipzap_6, field_g_value_zipzap_7)), true)
+  private val field_g = new StructField("g", new MapType(
+    new StructType(Arrays.asList(field_g_key_5)),
+    new StructType(Arrays.asList(field_g_value_zipzap)), true), true)
+  private val complexSchema =
```

Comment:
I'm not sure if this is the exact schema as you had used in the other example but maybe could you add in a comment the visual diagram of the schema?

It's hard to reason about these just looking at the variables

Reply:
Great callout, will do!